src/slice.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+import {Tensor, sizeOfShape} from './tensor.js';
+
+/**
+ * Produce a slice of the input tensor.
+ * @param {Tensor} input
+ * @param {Array} starts
+ * @param {Array} sizes
+ * @param {MLSliceOptions} options
+ * @return {Tensor}
+ */
+export function slice(input, starts, sizes, options = {}) {
+  /* Update the location with a given offset. */
+  function updateLocation(x, offset) {
+    const out = x.slice();
+    for (let i = 0; i < x.length; ++i) {
+      out[i] = x[i] + offset[i];
+    }
+    return out;
+  }
+
+  const rank = input.rank;
+  const startsForAllAxes = new Array(rank).fill(0);
+  let axes = options.axes;
+  if (axes) {
+    if (axes.length > rank) {
+      throw new Error(`The length of axes ${axes.length} is greater than rank ${rank}.`);
+    } else {
+      for (const axis of axes) {
+        if (!Number.isInteger(axis)) {
+          throw new Error(`Invalid axes value ${axis}, it should be an integer.`);
+        } else {
+          if (axis >= rank || axis < -rank) {
+            throw new Error(`Invalid axes value ${axis}, it should be in the interval ` +
+              `[${-rank}, ${rank}).`);
+          }
+        }
+      }
+    }
+  } else {
+    axes = [...Array(rank).keys()];
+  }
+  const axesLen = axes.length;
+  if (starts.length !== axesLen) {
+    throw new Error(`The length ${starts.length} of starts is not equal to the length ` +
+      `${axesLen} of axes.`);
+  }
+  if (sizes.length !== axesLen) {
+    throw new Error(`The length ${sizes.length} of sizes is not equal to the length ${axesLen} ` +
+      'of axes.');
+  }
+  const outputShape = input.shape.slice();
+  for (let i = 0; i < axesLen; ++i) {
+    const axis = axes[i] >= 0 ? axes[i] : axes[i] + rank;
+    const size = input.shape[axis];
+    const start = starts[i];
+    if (!Number.isInteger(start)) {
+      throw new Error(`Invalid starts value ${start}, it should be an integer.`);
+    }
+    startsForAllAxes[axis] = start >= 0 ? start : start + size;
+    if (start >= size || start < -size) {
+      throw new Error(`Invalid starts value ${start}, it shoule be in the interval ` +
+        `[${-size}, ${size}).`);
+    } else {
+      const slicedSize = sizes[i];
```

Comment:
size

Reply:
The variable `size` was used above for 
```js
const size = input.shape[axis];
```
so here needs another variable, thanks.

Reply:
@huningxin How about rename `size` to 'length'

```js
const length = input.shape[axis];
```

, then use `size` here?

```js
const size = sizes[i];
```

Reply:
Good point. I missed that. Then `sliceSize`, it means the size of the slice. Thanks.